main.bicep

```diff
@@ -589,5 +714,85 @@ resource notificationLogicApp 'Microsoft.Logic/workflows@2019-05-01' = {
   }
 }
 
+// this logic app unlike the other logic apps uses system-assigned identity, in order to assign Service Bus Sender Role
+resource univariateAnomalyDetectionLogicApp 'Microsoft.Logic/workflows@2019-05-01' = if(deployAnomalyDetectionResources) {
+  name: 'msdyn-iiot-sdi-uad-last-point-${uniqueIdentifier}'
+  location: resourcesLocation
+  identity: {
+    type: 'SystemAssigned'
+  }
+  properties: {
+    definition: json(loadTextContent('logic-apps/univariate-anomaly-detection-last-point.json')).definition
+    parameters: {
+      '$connections': {
+        value: {
+          servicebus: {
+            id: subscriptionResourceId('Microsoft.Web/locations/managedApis', resourcesLocation, 'servicebus')
+            connectionId: logicApp2ServiceBusConnection.id
+            connectionName: 'servicebus'
+            connectionProperties: {
+              authentication: {
+                type: 'ManagedServiceIdentity'
+              }
+            }
+          }
+        }
+      }
+      AnomalyDetectorIdentityAuthentication: {
+        value: {
+          audience: 'https://cognitiveservices.azure.com/'
+          type: 'ManagedServiceIdentity'
+        }
+      }
+      AnomalyDetectionEndpoint: {
+        value: anomalyDetector.properties.endpoint
+      }
+    }
+    accessControl: {
+      contents: {
+        allowedCallerIpAddresses: [
+          {
+            // See https://aka.ms/tmt-th188 for details.
+            addressRange: '0.0.0.0-0.0.0.0'
+          }
+        ]
+      }
+    }
+  }
+}
+
+resource serviceBusSenderRoleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = if(deployAnomalyDetectionResources) {
```

Comment:
Resource names here should probably convey who they are for. The what is already covered.

Sample: `anomalyDetectorServiceBusSenderRoleAssignment`

Reply:
Same applies to the below role assignments.

Reply:
Fixed